src/main/java/ch/njol/skript/expressions/ExprNumbers.java

```diff
@@ -81,10 +81,18 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 			s = f;
 			f = temp;
 		}
-		final double amount = integer ? Math.floor(f.doubleValue()) - Math.ceil(s.doubleValue()) + 1 : Math.floor(f.doubleValue() - s.doubleValue() + 1);
+		
 		final List<Number> list = new ArrayList<>();
+		
+		
 		final double low = integer ? Math.ceil(s.doubleValue()) : s.doubleValue();
-		for (int i = 0; i < amount; i++) {
+		double decimalCheck = s.doubleValue() % 1;
+		
+		final int decimals = decimalCheck == 0 ? 0 : Double.toString(decimalCheck).length() - 2;
+		
+		final double amount = integer ? Math.floor(f.doubleValue()) - Math.ceil(s.doubleValue()) + 1 : f.doubleValue() - s.doubleValue() + (10 ^ (decimals * -1));
+		
+		for (int i = 0; i < amount; i+= 10 ^ (decimals * -1)) {

```

Comment:
``amount`` is a double, but you're declaring ``i`` as an int, won't that cause rounding problems?

Reply:
You're right. Since this was the original code (then the increment was always 1, but now it can be 0.1, or 0.01, ...), I totally forgot about this.